publicodes/source/parseReference.ts

```diff
@@ -177,7 +176,12 @@ Par défaut, seul le premier s'applique. Si vous voulez un autre comportement, v
 			applicabilityEvaluation
 		)
 	}
-	const situationValue = situation[dottedName]
+	const situationValue = situation[dottedName]?.evaluate
```

Comment:
Pas convaincu que l'evaluation de la situation doit avoir lieu ici

Reply:
Oui je pense qu'elle peut être faite dans `publicode/index.ts` mais j'avais des problèmes sans cette modification. Il faut que je recreuse.

Reply:
J'ai mieux compris ce qui n'allait pas dans l'implémentation actuelle de l'évaluation de la situation, et qui m'a conduit a faire ce changement pas terrible comme tu l'as remarqué.

En fait actuellement la situation est évaluée avec `mapObjIndexed(value => evaluateExpression(value), situation)`, c'est à dire ligne par ligne. Or comme dans le reste de publicode, il faut évaluer les règles dans l'ordre topologique, et dans l'exemple de cette PR prendre en compte le temps partiel avant de calculer le SMIC (ce qui est à l'origine du bug que je rencontrais sans ce changement dans `parseReference.ts`

La solution me semble être de ne plus faire l'évaluation de la situation dans la méthode `setSituation` mais seulement le parsage résultants en un objet `this.parsedSituation` analogue à `this.parsedRules`, l'évaluation se faisait à la demande comme pour les règles normales.

Je vais essayer de faire la modification.

Reply:
Ca me semble effectivement une très bonne approche, qui a le mérite de ne plus rendre si différente l'évaluation de la règle avec situation ou avec formule. La situation étant en quelque sorte une formule de plus haute priorité.